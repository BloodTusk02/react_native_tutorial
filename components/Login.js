import React, { useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
} from "react-native";

const Login = () => {
  const [logText, setLogText] = useState("");
  const [passText, setPassText] = useState("");
  const [loginField, setLoginField] = useState({valid: false, value: "", error: ""});
  const [passwordField, setPasswordField] = useState({valid: false, value: "", error: ""});
  
  const submitForm = () => {
    let formValid = true;
    if (loginField.valid == false) {
      formValid = false;
      loginField.error = "Это поле обязательное для заполнения";
    } 
    if (passwordField.valid == false) {
      formValid = false;
      passwordField.error = "Это поле обязательное для заполнения";
    }

    if (formValid) {
      console.log("succ");
      return "form sent";
    }
    else {
      console.log("error");
      return "error";
    }
  }

  const validateRequired = (value) => {
    if (value) {
      return true;
    }
    return false;
  }

  return (
    <View style={styles.container}>
      <Text style={{ marginBottom: 20, fontSize: 25 }}>Login screen</Text>
      <View style={{ alignItems: "center" }}>
        <TextInput
          style={styles.input}
          placeholder="Email or Phone Number"
          onChangeText={(newText) => setLoginField({valid: validateRequired(newText), value: newText, error: ""})}
          defaultValue={loginField.value}
        />
        <Text style={{color: "#f00"}}>
          {loginField.error}
          </Text>
        <TextInput
          style={{ ...styles.input, marginVertical: 20 }}
          placeholder="Password"
          onChangeText={(newText) => setPasswordField({valid: validateRequired(newText), value: newText, error: ""})}
          defaultValue={passwordField.value}
        />
        <Text style={{color: "#f00"}}>{passwordField.error}</Text>
        <TouchableOpacity 
        style={styles.button} 
        onPress={() => {submitForm()}}>
          <Text style={styles.btnText}  >Sign In</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
    alignItems: "center",
    justifyContent: "center",
  },
  input: {
    height: 50,
    width: 290,
    paddingHorizontal: 10,
    fontSize: 20,
    borderWidth: 2,
    borderRadius: 10,
    borderColor: "#F57600",
  },
  button: {
    backgroundColor: "#F57600",
    height: 50,
    width: 290,
    alignItems: "center",
    justifyContent: "center",
    borderWidth: 2,
    borderRadius: 10,
    borderColor: "#F57600",
  },
  btnText: {
    color: "white",
    fontSize: 20,
  },
});

export default Login;
