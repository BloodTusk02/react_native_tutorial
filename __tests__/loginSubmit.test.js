import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../components/Login';


describe("Login", () => {
    it('invalid form', () => {
        const login = "";
        const password = "";
        console.log("OUTPUT");

        const setLoginField = jest.fn();
        const setPasswordField = jest.fn();
        const validateRequiredt = jest.fn();
        let loginScreen = renderer.create(<Login setLoginField={setLoginField} setPasswordField={setPasswordField} validateRequired={validateRequiredt}/>).root;

        console.log(loginScreen);
    
        const validateRequired = new loginScreen.validateRequired();
        loginScreen.setLoginField({valid: validateRequired(login), value: login, error: ""});
        loginScreen.setPasswordField({valid: validateRequired(password), value: password, error: ""});
        const result = loginScreen.submitForm();

        expect(result).toEqual("error");
      });
})
