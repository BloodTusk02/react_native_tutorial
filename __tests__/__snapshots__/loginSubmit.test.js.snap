// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginBottom": 20,
      }
    }
  >
    Login screen
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Email or Phone Number"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 20,
          "height": 50,
          "paddingHorizontal": 10,
          "width": 290,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#f00",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 20,
          "height": 50,
          "marginVertical": 20,
          "paddingHorizontal": 10,
          "width": 290,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#f00",
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F57600",
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 290,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
